Tribler/Test/Core/test_session.py

```diff
@@ -94,10 +95,13 @@ def test_add_torrent_def_to_channel(self):
         torrent_def = TorrentDef.load(TORRENT_FILE)
         extra_info = {"description": "iso"}
 
-        self.session.add_torrent_def_to_channel(channel_id, torrent_def, extra_info, forward=False)
+        @blocking_call_on_reactor_thread
+        def do_test():
+            self.session.add_torrent_def_to_channel(channel_id, torrent_def, extra_info, forward=False)
+        do_test()
```

Comment:
you can avoid this construct by using twisted.internet.threads.blockingCallFromThread()